### PR TITLE
(proof-new) Simplifications for proof rule checker interface

### PR DESCRIPTION
**Pull request by ajreynol**
Some of the interfaces in the proof rule checker are unnecessary now and can be deleted.

This also updates STRING_TRUST to have pedantic level 2.

### CODE REVIEW

File: src/theory/strings/proof_checker.cpp [hunk 1]
```diff
@@ -53,7 +53,7 @@ void StringProofRuleChecker::registerTo(ProofChecker* pc)
   pc->registerChecker(PfRule::STRING_CODE_INJ, this);
   pc->registerChecker(PfRule::STRING_SEQ_UNIT_INJ, this);
   // trusted rules
-  pc->registerTrustedChecker(PfRule::STRING_TRUST, this, 1);
+  pc->registerTrustedChecker(PfRule::STRING_TRUST, this, 2);
```
Comment (HanielB) [inline src/theory/strings/proof_checker.cpp:56]: Mention this change in the PR description as well

File: src/expr/proof_checker.cpp [hunk 2]
```diff
@@ -194,17 +161,20 @@ Node ProofChecker::checkDebug(PfRule id,
   // since we are debugging, we want to treat trusted (null) checkers as
   // a failure.
   Node res = checkInternal(id, cchildren, args, expected, out, false);
-  Trace(traceTag) << "ProofChecker::checkDebug: " << id;
-  if (res.isNull())
-  {
-    Trace(traceTag) << " failed, " << out.str() << std::endl;
-  }
-  else
+  if (Trace.isOn(traceTag))
```
Comment (HanielB) [inline src/expr/proof_checker.cpp:164]: Update the documentation of `checkDebug` to say that the check is done if we give an active trace.

**Review by HanielB — APPROVED** [review]
LGTM, just some minor doc requests

File: src/expr/proof_checker.cpp (hunk 2 above)
Comment (ajreynol) [reply src/expr/proof_checker.cpp:164]: Actually this was just an optimization, the traceTag doesn't impact whether the check is run (usually you are looking to see the return value).

**Pull request merged**